core/src/main/java/org/testcontainers/dockerclient/UnixSocketClientProviderStrategy.java

```diff
@@ -21,7 +25,12 @@
 
     @Override
     protected boolean isApplicable() {
-        return SystemUtils.IS_OS_LINUX;
+        return SystemUtils.IS_OS_LINUX || SystemUtils.IS_OS_MAC && new File(DOCKER_SOCK_PATH).exists();
```

Comment:
actually, just `IS_OS_UNIX` is enough. Changed. Also, there was a bug with `||` :D